### ISSUE TITLE
Helm3 hangs on modify deployed Prometheus-operator chart

### ISSUE THREAD

**Issue by fregatte** [helm/helm#6937]
Output of `helm version`:
version.BuildInfo{Version:"v3.0.0-rc.3", GitCommit:"2ed206799b451830c68bff30af2a52879b8b937a", GitTreeState:"clean", GoVersion:"go1.13.4"}

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.2", GitCommit:"f6278300bebbb750328ac16ee6dd3aa7d3549568", GitTreeState:"clean", BuildDate:"2019-08-05T09:23:26Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-eks-b7174d", GitCommit:"b7174db5ee0e30c94a0b9899c20ac980c0850fc8", GitTreeState:"clean", BuildDate:"2019-10-18T17:56:01Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Amazon EKS,
aws-cli/1.16.184 Python/2.7.15+ Linux/5.0.0-32-generic botocore/1.12.174

Got the problem after applying a new config change to the Prometheus Operator chart through `helm upgrade`. Helm hungs for a couple of minutes and returns the errors on any operations which related to release of PO chart:
```bash
$ helm upgrade mon charts/prometheus-operator/ -f values/dev/prometheus-operator/prometheus-operator.yml -v 8
I1112 15:54:39.334073    6120 request.go:968] Response Body: {"kind":"SecretList","apiVersion":"v1","metadata":{"selfLink":"/api/v1/namespaces/monitoring/secrets","resourceVersion":"21856695"},"items":[{"metadata":{"name":"mon.v57","namespace":"monitoring","selfLink":"/api/v1/namespaces/monitoring/secrets/mon.v57","uid":"f678f2dd-009e-11ea-913c-021caefde6b2","resourceVersion":"20695182","creationTimestamp":"2019-11-06T14:08:52Z","labels":{"modifiedAt":"1573049417","name":"mon","owner":"helm","status":"deployed","version":"57"}},"data":{"release":"SDRzSUFBQUFBQUFDLyt5OWUzT2l5djQzK2xhbTh1ZDUxc3dDakpsaHFuYlZFUk1Rb21TSnlxWFBmdXBYZEdNQWJaQWR2T0d2OW5zLzFRMG9HaTlna2xremE2aGFVN1dpMHQxMGY2K2Y3NlgvOXlhMGcvSE45NXRnRnQ3OGNlT0h6N09iNy85NzgreS94UFAvY2NZUm5pVmo1K2I3RGNldy9HZUcvOHpkRHRtNzd3enp2Zm50eTFmK0c4czNHZmJ1L3pDTjd3eHo4OGNOdG84OHhiS2ZtYnNoKy9VNzgrMTdrLzN5OVk3aE9QWmJZL3VRTThiak9mMDV3ekRzWi9yZmtHRyswLy9BelI4MzkrTVl2ZmpSM0orRk45OXZScEg3WWp2alQyZ1dST1RCbXo5dTRyazlYOFEzMzIrMlUvOXhFODdtWS9MUjBCdC8rdXRsRm96bjNuZ1JmM3FLeGkvMmZQYnl5YlBqVDNBOERqLzVZVHkzTVI0N1h6NjF2VEdhZnZMbjhhZDB4RTh3K2ZT [truncated 1462656 chars]
I1112 15:56:45.490964    6120 request.go:873] Stream error http2.StreamError{StreamID:0x3, Code:0x2, Cause:error(nil)} when reading response body, may be caused by closed connection.
Error: UPGRADE FAILED: query: failed to query with labels: stream error when reading response body, may be caused by closed connection. Please retry. Original error: stream error: stream ID 3; INTERNAL_ERROR
```

I get a similar message on any operation related to this release (upgrade, rollback, uninstall, list, etc). So, I cannot manage my PO installation now and even remove it from the kubernetes cluster. I've got similar errors earlier, but it was rare and operation could be implemented on repeat the command. For now, it doesn't work at all.

**Comment by bacongobbler** [comment]
From the error, it looks like the connection to the Kubernetes API server was closed unexpectedly during the upgrade. This could be a number of issues. Can you please check and observe the status of the API server during an upgrade?

**Comment by fregatte** [comment]
As I said, there is a problem only when I tried to modify the PO release. When I modified other charts or work with other resources it works fine.
I guess the problem was related to unsuccessful uninstall of the chart after which helm was thinking that it was alive but couldn't retrieve the information about the release for some reason. After a couple of iterations install/uninstall and removing helm secrets with releases' history I could successfully install the chart and that work for now. 
Thanks for the help!
